login-workflow/docs-new/screens/account-details.md

```diff
@@ -0,0 +1,72 @@
+# AccountDetailsScreen
+
+## Description
+A screen that displays text fields to collect the user's account details. The AccountDetailsScreen must be used in the context of the `RegistrationContextProvider`.
```

Comment:
Do we have a proper error message for this? If adopters use this outside the context provider, will it say "please use AccountDetailsScreen inside a RegistrationContextProvider"?

Reply:
You'll get a generic error when you try to read a value from a context that does not exist:

<img width="874" alt="image" src="https://github.com/etn-ccis/blui-react-workflows/assets/29152776/35858bdf-7760-4fc6-92b2-67d722473e69">

Reply:
If we want to be fancier than that, it would involve updating the hook to take a parameter describing the component that is making the call, e.g.:

```tsx
export const useRegistrationContext = (source?: string): RegistrationContextProviderProps => {
    const context = useContext(RegistrationContext);
    if (context === null) {
        throw new Error(`${source || 'useRegistrationContext'} must be used within an RegistrationContextProvider`);
    }
    return context;
};
```

and then update all of the places we use it:

```
useRegistrationContext('AccountDetailsScreen');
```

Reply:
That being said, we'll need to make distinctions between Full screens and Base screen components, since the later _can_ be used outside of the ContextProvider